### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by bquorning**
@libo

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -19,15 +19,10 @@ def self.total_capital_gain(year)
...
+    day = Date.new(year)
+    year_range = (day...day + 1.year)
 
-    day = Date.parse("1-1-#{year}")
-    beginning_of_year = day.at_beginning_of_year
-    end_of_year = day.at_end_of_year.to_date
-
-    where('executed_on >= ?', beginning_of_year)
-      .where('executed_on <= ?', end_of_year)
+    where(executed_on: year_range)
```
Comment (libo) [inline app/models/event.rb]: sure it works on Postgres?

Comment (libo) [inline app/models/event.rb]: and mysql?

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -49,7 +49,12 @@
 </table>
 
 <%= form_tag(events_path, method: :get) do %>
-  <%= select_year(@year, field_name: 'year') %>
+  <%= select_year(
+    @year,
+    field_name: 'year',
+    start_year: @events.first.executed_on.year,
+    end_year: @events.last.executed_on.year
+  ) %>
```
Comment (libo) [inline app/views/events/index.html.erb]: :heart: eheh!

File: app/models/event.rb (hunk 1 above)
Comment (bquorning) [inline app/models/event.rb]: Section 2.3.2 of http://guides.rubyonrails.org/active_record_querying.html#hash-conditions

--
Benjamin

**Comment by libo** [comment]
:+1:

**Pull request merged**